include/h2o/http1client.h

```diff
@@ -46,7 +46,7 @@ typedef struct st_h2o_http1client_header_t {
 typedef int (*h2o_http1client_body_cb)(h2o_http1client_t *client, const char *errstr);
 typedef h2o_http1client_body_cb (*h2o_http1client_head_cb)(h2o_http1client_t *client, const char *errstr, int minor_version,
                                                            int status, h2o_iovec_t msg, h2o_http1client_header_t *headers,
-                                                           size_t num_headers);
+                                                           h2o_str_case_t **orig_hname_case, size_t num_headers);
```

Comment:
How about changing the argument: `h2o_http1client_header_t *headers` to `h2o_header_t *`?

Doing so, and tokenizing the names right after [calling `h2o_strtolower`](https://github.com/h2o/h2o/blob/master/lib/common/http1client.c#L255) seems like the way we should go.

By doing so, costly comparisons of iterating through the headers in the later moments in the `on_head` function of lib/common/http1client.c can be optimized to use token comparisons (e.g. change `h2o_memis(headers[i].name, headers[i].name_len, H2O_STRLIT("connection"))` to headers[i].name == H2O_TOKEN_CONNECTION).

I do not think the additional amount of stack used by the list of `h2o_header_t` would be a issue, if we could leave the scope that allocates `struct phr_header headers[100]` before calling `on_head`.